src/ray/core_worker/core_worker.cc

```diff
@@ -72,7 +72,9 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
       heartbeat_timer_(io_service_),
       worker_server_(WorkerTypeString(worker_type), 0 /* let grpc choose a port */),
       gcs_client_(gcs_options),
-      memory_store_(std::make_shared<CoreWorkerMemoryStore>()),
+      reference_counter_(std::make_shared<ReferenceCounter>()),
+      memory_store_(std::make_shared<CoreWorkerMemoryStore>(
+          RayConfig::instance().ref_counting_enabled() ? reference_counter_ : nullptr)),
```

Comment:
Could you enable this only for python workers? aka, only when worker language is python.

Because java doesn't implement ref counting yet, enabling this for java would cause memory leak in this case.

Reply:
Thanks, fixed this.